scrnaseq.Rmd

```diff
@@ -342,155 +506,341 @@ Finally, __cell cycle scores__ are assigned to each cell based on its normalised
 
 For each gene, we calculate its median expression across all cells, and then calculate the deviation from this median for each cell. For each cell, we plot the median expression (black), the interquartile range (lightgrey), whiskers defined as 1.5 times the interquartile range (darkgrey), and outliers (`r param$col`). 
 
-### Raw counts
-```{r plot_norm_raw}
-raw = log2(raw + 1)
-p = PlotRLE(raw, col=param$col) + labs(title="log2(raw counts + 1)")
+#### Raw counts
+```{r plot_RLE_raw}
+# Get counts from assay RNA
+# Note: The counts are retrieved as sparse matrices and then cbind is used to combine them. Only at the plotRLE,
+cell_subset_raw = purrr::map(seq(length(sc)), function(i) {
+  d = GetAssayData(subset(sc[[i]], cells=cell_subset_ids[[i]], features=shared_rna_features), assay="RNA", slot="counts")
+  colnames(d) = paste(colnames(d), i, sep=".")
+  return(d)
+})
+cell_subset_raw = purrr::reduce(cell_subset_raw, cbind)
+
+# Plot
+p = PlotRLE(as.matrix(log2(cell_subset_raw + 1)), id=cell_identities, col=param$col_samples) + labs(title="log2(raw counts + 1)")
 p
 ```
 
-### Standard normalisation
-```{r plot_norm_norm}
-p = PlotRLE(norm1, col=param$col) + labs(title="Normalised data")
+#### Standard normalisation (RNA)
+```{r plot_RLE_RNA}
+# Get normalised data from assay RNA
+# Note: The counts are retrieved as sparse matrices and then cbind is used to combine them
+cell_subset_norm1 = purrr::map(seq(length(sc)), function(i) {
+  d = GetAssayData(subset(sc[[i]], cells=cell_subset_ids[[i]], features=shared_rna_features), assay="RNA", slot="data")
+  colnames(d) = paste(colnames(d), i, sep=".")
+  return(d)
+})
+cell_subset_norm1 = purrr::reduce(cell_subset_norm1, cbind)
+
+# Plot
+p = PlotRLE(as.matrix(cell_subset_norm1), id=cell_identities, col=param$col_samples) + labs(title="Normalised data")
 p 
 ```
 
-### Advanced normalisation (SCT)
-```{r plot_norm_sct}
-p = PlotRLE(norm2, col=param$col) + labs(title="SCTransform'ed data")
+#### Advanced normalisation (SCT)
+```{r plot_RLE_SCT}
+# Get normalised data from assay SCT
+# Note: The counts are retrieved as sparse matrices and then cbind is used to combine them
+cell_subset_norm2 = purrr::map(seq(length(sc)), function(i) {
+  d = GetAssayData(subset(sc[[i]], cells=cell_subset_ids[[i]], features=shared_sct_features), assay="SCT", slot="data")
+  colnames(d) = paste(colnames(d), i, sep=".")
+  return(d)
+})
+cell_subset_norm2 = purrr::reduce(cell_subset_norm2, cbind)
+
+# Plot
+p = PlotRLE(as.matrix(cell_subset_norm2), id=cell_identities, col=param$col_samples) + labs(title="SCTransform'ed data")
 p 
 ```
 
+<!-- https://stackoverflow.com/questions/38062706/rmarkdown-how-to-end-tabbed-content -->
```

Comment:
Interesting. What does this syntax do?

Reply:
It is just a comment in HTML and is the link to the site where I found the piece of code about </div>. It will not be shown.